### DISCUSSION TITLE
how do you solve error 401 in your application

### DISCUSSION THREAD

**Discussion by dodicandra** [TanStack/query#2389] [Q&A]
I'm having a little trouble with this, let's say it's related to authentication, if your app doesn't have a refresh token feature, when this token expires what you want is to re-login your app, let's say we store user credentials in the app and reuse them for login. repeat. if this is successful, what is the proper way to redo the previous request where the error could be a `POST` or `GET` request together with the data sent?

I want to hear your opinion on cases like this

**Comment by dusty** [comment]
We are currently checking the expiration of the jwt before the actual request is fired.  If its about to expire, and we have a refresh token available, then we call authentication service with a refresh token then use the new access token in the original request.  Being proactive there.

You can also pass in a function to the retry config option, so if the request fails it will call that function to determine if it should retry.  You could take the opportunity there to get a new token before attempting a retry.

**Reply by dusty** [reply]
Sorry totally misread, I think you were saying you don't have a refresh token.  I suppose if you are actually storing the credentials, you could use the same behavior to accomplish what you want.

**Reply by dodicandra** [reply]
yes it might be possible to try with the same behavior, but is it possible to perform a re-login action, and repeat failed requests after token expiration?

I'm a little confused on how to implement it, do you have a little code snippet as an illustration?
